pkgs/by-name/mf/mfem/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  gfortran,
+  mpi,
+  mpiSupport ? true,
+  mpiCheckPhaseHook,
+  blas,
+  hypre,
+  metis,
+  suitesparse,
+  withSuiteSparse ? mpiSupport,
+  mumps_par,
+  withMumps ? mpiSupport,
+}:
+assert withSuiteSparse -> mpiSupport;
+assert withMumps -> mpiSupport;
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mfem";
+  version = "4.7";
+
+  src = fetchFromGitHub {
+    owner = "mfem";
+    repo = "mfem";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-NJI8ABsfnnzMETfyscpNX8qK72BFF9FfxsY0J48aRtg=";
+  };
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+  ] ++ lib.optional mpiSupport mpi;
+
+  buildInputs =
+    lib.optionals mpiSupport [
+      blas
+      hypre # mpi->hypre
+      metis # hypre->metis
```

Comment:
What does the arrow mean?

Reply:
mpi requires hypre and hypre requires metis in cmakefiles

Reply:
In cmakefiles of whom?

Reply:
of mfem: https://github.com/mfem/mfem/blob/17955e114020af340e9a06a66ebef43e05012d9c/CMakeLists.txt#L260